cluster/src/main/java/io/scalecube/cluster/gossip/GossipProtocol.java

```diff
@@ -48,35 +48,45 @@
 public final class GossipProtocol implements IGossipProtocol, IManagedGossipProtocol {
   private static final Logger LOGGER = LoggerFactory.getLogger(GossipProtocol.class);
 
-  private ITransport transport;
+  private final String memberId;
+  private final ITransport transport;
+  private final ScheduledExecutorService executor;
```

Comment:
It make sense, but there are more issues with executor. I think it deserves separate PR.